miniapp/miniapp_cholesky.cpp

```diff
@@ -141,15 +154,15 @@ int hpx_main(hpx::program_options::variables_map& vm) {
       std::cout << "[" << run_index << "]"
                 << " " << elapsed_time << "s"
                 << " " << gigaflops << "GFlop/s"
-                << " " << matrix.size() << " " << matrix.blockSize() << " " << comm_grid.size() << " "
-                << hpx::get_os_thread_count() << std::endl;
+                << " " << matrix_host.size() << " " << matrix_host.blockSize() << " " << comm_grid.size()
+                << " " << hpx::get_os_thread_count() << std::endl;
 
     // (optional) run test
     if ((opts.do_check == CholCheckIterFreq::Last && run_index == (opts.nruns - 1)) ||
         opts.do_check == CholCheckIterFreq::All) {
-      MatrixType original(matrix_size, block_size, comm_grid);
+      HostMatrixType original(matrix_size, block_size, comm_grid);
       copy(matrix_ref, original);
-      check_cholesky(original, matrix, comm_grid);
+      check_cholesky(original, matrix_host, comm_grid);
```

Comment:
Don't we need a `matrix.syncTargetToSource()` before checking if the result is correct?

Reply:
It's synchronized in the destructor, when possible (depends on constness; in this case it's possible). Do you think having it explicit outside of the destructor would be clearer?

Reply:
I didn't notice that a copy occurs upon destruction. Thanks for clarifying, I find it clear enough as it is.